packages/flutter/lib/src/material/reorderable_list.dart

```diff
@@ -581,9 +590,10 @@ class _ReorderableListContentState extends State<_ReorderableListContent> with T
         reverse: widget.reverse,
         child: _buildContainerForScrollDirection(
           children: <Widget>[
-            if (widget.reverse) _wrap(finalDropArea, widget.children.length, constraints),
+            if (widget.reverse) _wrap(finalDropArea, 0, constraints),
```

Comment:
Could you explain this change? I'm not clear what this does

Reply:
This fixes https://github.com/flutter/flutter/issues/51359.

Reply:
I've added a comment explaining why it must be 0.

Reply:
We should make this fix a separate pull request. I think it starts to get confusing what the goal of this PR is when you start adding fixes to non-related issues. You would also have to add a test that verifies that this behavior has been fixed so that the framework does not regress. 

Could you either revert this and create a new PR for this fix, or add a test here and change the PR title and description?

Reply:
I've reverted the unrelated changed and made a new PR for them.